src/NHibernate/AdoNet/GenericBatchingBatcher.cs

```diff
@@ -0,0 +1,235 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Common;
+using System.Text;
+using NHibernate.AdoNet.Util;
+using NHibernate.Exceptions;
+using NHibernate.SqlCommand;
+
+namespace NHibernate.AdoNet
+{
+	/// <summary>
+	/// A generic batcher that will batch UPDATE/INSERT/DELETE commands by concatenating them with a semicolon.
+	/// Use this batcher only if there are no dedicated batchers in the given environment. Unfortunately some
+	/// database clients do not support concatenating commands with a semicolon. Here are the known clients
+	/// that do not work with this batcher:
+	/// - FirebirdSql.Data.FirebirdClient
+	/// - Oracle.ManagedDataAccess
+	/// - System.Data.SqlServerCe
+	/// </summary>
+	public partial class GenericBatchingBatcher : AbstractBatcher
+	{
+		private readonly int? _maxNumberOfParameters;
+		private readonly BatchingCommandSet _currentBatch;
+		private int _totalExpectedRowsAffected;
+		private StringBuilder _currentBatchCommandsLog;
+
+		public GenericBatchingBatcher(ConnectionManager connectionManager, IInterceptor interceptor, char statementTerminator)
+			: base(connectionManager, interceptor)
+		{
+			BatchSize = Factory.Settings.AdoBatchSize;
+			StatementTerminator = statementTerminator;
```

Comment:
It is possible to handle this as a setting, read through `Factory.Settings`. But it would have been better to have access to the configuration properties dictionary rather than adding a new property to `Settings` for such a specialized parameter. Unfortunately it seems not available.

So maybe is it better to keep current implementation as it is currently.

Reply:
Should come from the dialect.

Reply:
This thing has come from a suggestion of mine. I was stating `;` is the standard, followed by all databases I have checked, but with some allowing it to be changed to something else. So I had written:

> Since at least some databases allow to redefine the statement terminaison character to something else than a semi-colon, maybe this character should be configurable. But well, it would be just a small nice to have.

Putting it in the dialect sounds indeed better. An overridable property in `BaseDialect` should be enough, those needing an other character would then be able of having it with a custom dialect. (If the need is frequent users could then ask for a configuration property, since the dialects are configurable.)

Reply:
Moved into `BaseDialect` and used it also in `BasicResultSetsCommand`, when batching select statements.